lib/membrane_framerate_converter.ex

```diff
@@ -134,17 +135,17 @@ defmodule Membrane.FramerateConverter do
     if state.target_pts > input_buffer.pts do
       state = %{state | last_buffer: input_buffer}
 
-      {Enum.reverse(buffers), state}
+      {Enum.reverse(buffers), %{state | caps_changed?: false}}
     else
       last_buffer = state.last_buffer
       dist_right = input_buffer.pts - state.target_pts
       dist_left = state.target_pts - last_buffer.pts
 
       new_buffer =
-        if dist_left <= dist_right do
-          %{last_buffer | pts: state.target_pts}
-        else
+        if dist_left >= dist_right or state.caps_changed? do
```

Comment:
I'd add a comment here explaining the logic, e.g.
```suggestion
        # If caps have changed, use only the new buffer
        if dist_left >= dist_right or state.caps_changed? do
```